File: AllegroDotNet.Wrapper/src/AllegroEventSource.cpp
#include "PCH.h"

#include "AllegroEventSource.h"

namespace AllegroDotNet::Wrapper
{
	AllegroEventSource::AllegroEventSource()
		: NativePointer(new ALLEGRO_EVENT_SOURCE())
	{
	}

	AllegroEventSource::AllegroEventSource(ALLEGRO_EVENT_SOURCE* nativeEventSource)
		: NativePointer(nativeEventSource)
	{
	}
}

File: Source/AllegroDotNet/Models/AllegroEventSource.cs
using System.Runtime.InteropServices;

namespace SubC.AllegroDotNet.Models
{
  public sealed class AllegroEventSource : NativePointerModel
  {
    //internal NativeAllegroEventSource NativeEventSource = new();

    [StructLayout(LayoutKind.Sequential)]
    internal struct NativeAllegroEventSource
    {
      [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
      int[] __pad;
    }
  }
}